<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>422</width>
    <height>534</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="windowIcon">
   <iconset resource="../icons.qrc">
    <normaloff>:/images/application.svg</normaloff>:/images/application.svg</iconset>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout">
    <item>
     <widget class="QGroupBox" name="groupStatus">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="styleSheet">
       <string notr="true">QGroupBox {background: right bottom no-repeat url(:/images/icons/dialog-warning.svg)}</string>
      </property>
      <property name="title">
       <string>Status</string>
      </property>
      <layout class="QGridLayout" name="gridLayout_2">
       <item row="0" column="0">
        <widget class="QLabel" name="label">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string>Asynch. Read:</string>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="label_2">
         <property name="text">
          <string>Memory Status:</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QLabel" name="labelMemoryStatusValue">
         <property name="text">
          <string>?</string>
         </property>
        </widget>
       </item>
       <item row="0" column="3">
        <spacer name="horizontalSpacer_2">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="0" column="1">
        <widget class="QLabel" name="labelAsynchValue">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string>?</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QGroupBox" name="groupTrigger">
      <property name="styleSheet">
       <string notr="true"/>
      </property>
      <property name="title">
       <string>Trigger</string>
      </property>
      <property name="flat">
       <bool>false</bool>
      </property>
      <property name="checkable">
       <bool>false</bool>
      </property>
      <layout class="QGridLayout" name="gridLayout"/>
     </widget>
    </item>
    <item>
     <widget class="QGroupBox" name="groupBoxVCD">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="styleSheet">
       <string notr="true"/>
      </property>
      <property name="title">
       <string>Sample download and VCD-Export</string>
      </property>
      <layout class="QGridLayout" name="gridLayout_3">
       <property name="rightMargin">
        <number>9</number>
       </property>
       <property name="bottomMargin">
        <number>9</number>
       </property>
       <item row="1" column="1" colspan="2">
        <widget class="QPushButton" name="buttonDownload">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="layoutDirection">
          <enum>Qt::LeftToRight</enum>
         </property>
         <property name="styleSheet">
          <string notr="true">align: right</string>
         </property>
         <property name="text">
          <string>&amp;Download</string>
         </property>
         <property name="flat">
          <bool>false</bool>
         </property>
        </widget>
       </item>
       <item row="0" column="2">
        <widget class="QToolButton" name="toolButtonVCD">
         <property name="text">
          <string>...</string>
         </property>
        </widget>
       </item>
       <item row="0" column="0">
        <widget class="QLabel" name="labelVCD">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string>VCD-File:</string>
         </property>
         <property name="textFormat">
          <enum>Qt::PlainText</enum>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QLineEdit" name="lineEditVCD"/>
       </item>
      </layout>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>422</width>
     <height>25</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuLogic_Analyzer">
    <property name="title">
     <string>Logic Analyzer</string>
    </property>
    <widget class="QMenu" name="menuTPG">
     <property name="statusTip">
      <string>Override hardware's inputs with test pattern</string>
     </property>
     <property name="title">
      <string>Test Pattern Generator</string>
     </property>
     <property name="icon">
      <iconset resource="../icons.qrc">
       <normaloff>:/images/tpg.svg</normaloff>:/images/tpg.svg</iconset>
     </property>
     <addaction name="menuTPG_Off"/>
     <addaction name="menuTPG_Pattern1"/>
     <addaction name="menuTPG_Pattern2"/>
     <addaction name="menuTPG_Pattern3"/>
    </widget>
    <widget class="QMenu" name="menuTT">
     <property name="title">
      <string>Trigger Timing</string>
     </property>
     <addaction name="menuTT_Pre"/>
     <addaction name="menuTT_Center"/>
     <addaction name="menuTT_Post"/>
    </widget>
    <addaction name="actionShow_Config"/>
    <addaction name="separator"/>
    <addaction name="actionRun_Length_Encoding"/>
    <addaction name="menuTT"/>
    <addaction name="menuTPG"/>
    <addaction name="separator"/>
    <addaction name="menuSoftReset"/>
    <addaction name="menuColdReset"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuLogic_Analyzer"/>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
  <action name="actionExit">
   <property name="text">
    <string>Quit</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Q</string>
   </property>
   <property name="menuRole">
    <enum>QAction::QuitRole</enum>
   </property>
  </action>
  <action name="menuSoftReset">
   <property name="icon">
    <iconset resource="../icons.qrc">
     <normaloff>:/images/refresh.svg</normaloff>:/images/refresh.svg</iconset>
   </property>
   <property name="text">
    <string>Reset Sampling Pipeline</string>
   </property>
   <property name="statusTip">
    <string>Keep settings and prepare for new sampling</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+R</string>
   </property>
   <property name="iconVisibleInMenu">
    <bool>true</bool>
   </property>
  </action>
  <action name="menuColdReset">
   <property name="icon">
    <iconset resource="../icons.qrc">
     <normaloff>:/images/reset.svg</normaloff>:/images/reset.svg</iconset>
   </property>
   <property name="text">
    <string>Restore Default Settings</string>
   </property>
   <property name="statusTip">
    <string>Restore default settings and prepare for new samling</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+R</string>
   </property>
   <property name="iconVisibleInMenu">
    <bool>true</bool>
   </property>
  </action>
  <action name="menuTPG_Off">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Off</string>
   </property>
  </action>
  <action name="menuTPG_Pattern1">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Pattern 1</string>
   </property>
  </action>
  <action name="menuTPG_Pattern2">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Pattern 2</string>
   </property>
  </action>
  <action name="menuTPG_Pattern3">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Pattern 3</string>
   </property>
  </action>
  <action name="actionNot_Support_by_GUI_yet">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Not Support by GUI yet</string>
   </property>
  </action>
  <action name="actionShow_Config">
   <property name="icon">
    <iconset resource="../icons.qrc">
     <normaloff>:/images/show-config.svg</normaloff>:/images/show-config.svg</iconset>
   </property>
   <property name="text">
    <string>Show Hardware Configuration</string>
   </property>
   <property name="statusTip">
    <string>Show configuration recieved from hardware</string>
   </property>
   <property name="iconVisibleInMenu">
    <bool>true</bool>
   </property>
  </action>
  <action name="actionRun_Length_Encoding">
   <property name="enabled">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Run Length Encoding</string>
   </property>
   <property name="statusTip">
    <string>Not supported by GUI yet</string>
   </property>
  </action>
  <action name="menuTT_Pre">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Pre Trigger</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+1</string>
   </property>
  </action>
  <action name="menuTT_Center">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Center Triggering</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+2</string>
   </property>
  </action>
  <action name="menuTT_Post">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Post Triggering</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+3</string>
   </property>
  </action>
 </widget>
 <resources>
  <include location="../icons.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>toolButtonVCD</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>vcdFileChooser()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>641</x>
     <y>478</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>266</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionShow_Config</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>showConfig()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>266</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>menuSoftReset</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>sendSoftReset()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>266</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>menuColdReset</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>sendColdReset()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>266</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>actionExit</sender>
   <signal>triggered()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>-1</x>
     <y>-1</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>266</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonDownload</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>vcdDownloadFile()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>738</x>
     <y>478</y>
    </hint>
    <hint type="destinationlabel">
     <x>399</x>
     <y>266</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>vcdDownloadFile()</slot>
  <slot>showConfig()</slot>
  <slot>sendSoftReset()</slot>
  <slot>sendColdReset()</slot>
  <slot>refreshLAStatus()</slot>
  <slot>loadLASettings()</slot>
  <slot>vcdFileChooser()</slot>
 </slots>
</ui>
